src/java.base/share/classes/java/io/Reader.java

```diff
@@ -391,6 +397,71 @@ public int read(char[] cbuf) throws IOException {
      */
     public abstract int read(char[] cbuf, int off, int len) throws IOException;
 
+    /**
+     * Reads all remaining characters as lines of text. A line is considered to
+     * be terminated by any one of a line feed ('\n'), a carriage return ('\r'),
+     * a carriage return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
```

Comment:
I think the definition here should be copied from String.lines(), or it should refer to it. The main difference is that spec is explicit about a line being zero-or-more chars followed by a line terminator or one-or-more chars followed by EOF. This implies there is no blank line between a trailing line terminator and EOF.